docs/guides/chip_tool_guide.md

```diff
@@ -0,0 +1,769 @@
+    step 4
+
+#### Commissioning into Wi-Fi network over Bluetooth LE
+
+```
+./BUILD_PATH/chip-tool pairing ble-wifi <node_id> <ssid> <password> <pin_code> <discriminator>
+
+```
+
+where:
+
+-   _<node_id\>_ is the user-defined ID of the node being commissioned,
+-   _<ssid\>_ and _<password\>_ are credentials determined in step 3,
+-   _<pin_code\>_ and _<discriminator\>_ are device specific keys determined in
+    step 4
+
+If the hexadecimal format is preferred the `hex:` prefix shall be used, i.e:
+
+```
+./BUILD_PATH/chip-tool pairing ble-wifi <node_id> hex:<ssid> hex:<password> <pin_code> <discriminator>
+
+```
+
+The _<node_id>_ can be simply provided as a hexadecimal value with `0x` prefix.
+
+After connecting the device over Bluetooth LE, the controller will go through
+the following stages:
+
+-   Establishing a secure connection that completes the PASE
+    (Password-Authenticated Session Establishment) session using SPAKE2+
+    protocol and results in printing the following log:
+
+        ```
+        Secure Session to Device Established
+        ```
+
+-   Providing the device with a network interface using ZCL Network
+    Commissioning cluster commands, and the network pairing credentials from
+    step 3
+-   Discovering the IPv6 address of the Matter accessory using the SRP (Service
+    Registration Protocol) for Thread devices, or the mDNS (Multicast Domain
+    Name System) protocol for Wi-Fi or Ethernet devices. It results in printing
+    log that indicates that the node address has been updated. The IPv6 address
+    of the device is cached in the controller for later usage.
+-   Closing the Bluetooth LE connection, as the commissioning process is
+    finished and the `chip-tool` is now using only the IPv6 traffic to reach the
+    device.
+
+#### Commissioning over IP
+
+The command below will discover devices and try to pair with the first one it
+discovers using the provided _setup PIN code_.
+
+```
+./BUILD_PATH/chip-tool pairing onnetwork <node_id> <pin_code>
+
+```
+
+where:
+
+-   _<node_id\>_ is the user-defined ID of the node being commissioned,
+-   _<pin_code\>_ is device specific _setup PIN code_ determined in step 4
+
+The command below will discover devices with long discriminator and try to pair
+with the first one it discovers using the provided setup code.
+
+```
+./BUILD_PATH/chip-tool pairing onnetwork-long <node_id> <pin_code> <discriminator>
+
+```
+
+where:
+
+-   _<node_id\>_ is the user-defined ID of the node being commissioned,
+-   _<pin_code\>_ and _<discriminator\>_ are device specific keys determined in
+    step 4
+
+The command below will discover devices based on the given QR code payload
+(which devices log when they start up) and try to pair with the first one it
+discovers.
+
+```
+./BUILD_PATH/chip-tool qrcode <node_id> <qrcode_payload>
+
+```
+
+where:
+
+-   _<node_id\>_ is the user-defined ID of the node being commissioned,
+-   _<qrcode_payload\>_ is the QR code payload
```

Comment:
My concern for this section: are these commands I need to run one after another? Do I need to run only one depending on the scenario? If only one, mention the one, and let's create a section at the bottom of the page with "available commissioning commands" or something like this that lists all relevant commands...

Reply:
All commands from `Commissioning over IP` are standalone means to commission the device over IP - and the commissioning over IP and other commissioning methods (Thread/WiFi) are mutually exclusive (right, @Damian-Nordic?). Maybe I will just separate `Commissioning over IP` methods into dedicated sections and add a general description in to the main section (`Commissioning over IP`).